### PR TITLE
Allow local media to be marked as safe from being quarantined

### LINKED ISSUE #4988: /admin/quarantine_media destroys stickerpacks
we need a way to exclude stickerpacks from being nuked by quarantine_media somehow. suggestions welcome

### PR DESCRIPTION
**Pull request by clokep**
Should be reviewable commit by commit, but this:
* Combines some logic that was duplicated (can split this to a separate PR if it makes it easier).
* Adds a column to `local_media_repository` to keep a piece of media from being quarantined.
* Modifies the quarantine logic to use the above flag.
* Adds tests.

Fixes #4988

### CODE REVIEW

**Comment by clokep** [comment]
Note that there's no way to set this right now (e.g. no admin API), and no way to check the status of this flag...is that OK right now?

File: synapse/storage/data_stores/main/media_repository.py [hunk 1]
```diff
@@ -81,6 +81,15 @@ def store_local_media(
             desc="store_local_media",
         )
 
+    def mark_local_media_as_safe(self, media_id: str):
+        """Mark a local media as safe from quarantining."""
+        return self.db.simple_update_one(
+            "local_media_repository",
+            {"media_id": media_id},
+            {"safe_from_quarantine": True},
```
Comment (erikjohnston) [inline synapse/storage/data_stores/main/media_repository.py:89]:
```suggestion
            table="local_media_repository",
            keyvalues={"media_id": media_id},
            updatevalues={"safe_from_quarantine": True},
```

File: tests/rest/admin/test_admin.py [hunk 2]
```diff
@@ -480,10 +473,10 @@ def test_quarantine_all_media_by_user(self):
 
         # Should be quarantined
```
Comment (erikjohnston) [inline tests/rest/admin/test_admin.py:474]:
```suggestion
        # Shouldn't be quarantined
```

File: tests/rest/admin/test_admin.py [hunk 3]
```diff
@@ -480,10 +473,10 @@ def test_quarantine_all_media_by_user(self):
...
         self.assertEqual(
-            404,
+            200,
             int(channel.code),
             msg=(
-                "Expected to receive a 404 on accessing quarantined media: %s"
+                "Expected to receive a 200 on accessing quarantined media: %s"
```
Comment (erikjohnston) [inline tests/rest/admin/test_admin.py:479]:
```suggestion
                "Expected to receive a 200 on accessing not-quarantined media: %s"
```

**Pull request merged**